packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.tsx

```diff
@@ -127,6 +127,28 @@ export const useTreeViewItems: TreeViewPlugin<UseTreeViewItemsSignature> = ({
     [store],
   );
 
+  const setIsItemDisabled = useEventCallback(
+    ({ itemId, shouldBeDisabled }: { itemId: string; shouldBeDisabled?: boolean }) => {
+      store.update((prevState) => {
+        const itemMetaLookup = { ...prevState.items.itemMetaLookup };
+        if (itemMetaLookup[itemId]) {
+          itemMetaLookup[itemId] = {
+            ...itemMetaLookup[itemId],
+            disabled: shouldBeDisabled || !itemMetaLookup[itemId].disabled,
```

Comment:
```suggestion
            disabled: shouldBeDisabled ?? !itemMetaLookup[itemId].disabled,
```

Otherwise, if `shouldDIsabled === false` and the item is already not disabled, it would disable it :laughing:

Reply:
General rule of thumb, never use `||` to apply a default value, always use `??`
Only use `||` for boolean rules (in a `if` statement most of the time)